lib/ui/widgets/cheader.dart

```diff
@@ -156,8 +160,10 @@ class CHeader {
                   textStyle: themeData.textTheme.headline3
                       .copyWith(color: CColors.white),
                   borderRadius: BorderRadius.circular(28),
-                  onPressed: () {
-                    Navigator.pushAndRemoveUntil(
+                  onPressed: () async {
+                    await _authRepository.logout();
+
+                    await Navigator.pushAndRemoveUntil(
                       context,
                       MaterialPageRoute(builder: (context) => WelcomeScreen()),
```

Comment:
Maybe we can use named route check `routes.dart`
**Navigator.pushNamed(context, '/home');**